migrations/src/main/scala/za/co/absa/enceladus/migrations/continuous/migrate01/MigratorDataset.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.migrations.continuous.migrate01
+
+import java.time.ZonedDateTime
+
+import org.apache.log4j.{LogManager, Logger}
+import org.mongodb.scala.MongoDatabase
+import za.co.absa.enceladus.migrations.continuous.EntityVersionMap
+import za.co.absa.enceladus.migrations.framework.ObjectIdTools
+import za.co.absa.enceladus.migrations.migrations.MigrationToV1.convertConformanceRule
+import za.co.absa.enceladus.migrations.migrations.model0.Serializer0
+import za.co.absa.enceladus.migrations.migrations.model1.Serializer1
+import za.co.absa.enceladus.migrations.migrations.{model0, model1}
+
+import scala.util.control.NonFatal
+
+/**
+  * The class provides datasets continuous migration from model version 0 to model version 1.
+  *
+  * @param evm         An entity mapper for tracking the mapping between old versions of entities and new ones.
+  * @param databaseOld An instance of a MongoDB database connection containing old model documents.
+  * @param databaseNew An instance of a MongoDB database connection containing new model documents.
+  */
+final class MigratorDataset(evm: EntityVersionMap,
```

Comment:
```suggestion
final class DatasetMigrator(evm: EntityVersionMap,
```
Same as for `SchemaMigrator`, probably for `MappngTable` and `Run` as well

Reply:
Usually, I like common (more general) name bits to go at the beginning of a name so files form a natural grouping. E.g. all migrator subclasses are grouped together in the list of files or the list of classes. But it might be just a habit from C++ where tooling for managing classes is not so great. Agree that reversing the bits sounds more natural.